driver/src/contracts/stablex_contract.rs

```diff
@@ -108,6 +125,38 @@ impl StableXContract for StableXContractImpl {
         Ok(Duration::from_secs(remaining_seconds.as_u64()))
     }
 
+    fn get_filtered_auction_data_paginated(
+        &self,
+        batch_index: U256,
+        page_size: u16,
+        previous_page_user: Address,
+        previous_page_user_offset: u16,
+    ) -> Result<FilteredAuctionData> {
+        let target_batch = batch_index.low_u32();
+        let mut builder = self.viewer.get_filtered_orders_paginated(
+            [target_batch, target_batch, target_batch],
+            vec![],
+            previous_page_user,
+            previous_page_user_offset,
+            page_size,
+        );
+        builder.m.tx.gas = None;
+        builder
+            .call()
+            .wait()
+            .map(
```

Comment:
I'm curious why you prefer `map` and `map_err` to `?` and then using the return value in the function?

I tend to prefer `?` as it creates less "rightwards drift".

Reply:
No reason, just the way it was done in this class before. Will change!